gocd/resource_pipeline_stage_test.go

```
@@ -120,12 +204,32 @@ func testCheckPipelineStageExists(resource string) r.TestCheckFunc {
 			return fmt.Errorf("No pipeline stage name is set")
 		}
 
+		cfg := gocd.Configuration{
+			Server:   os.Getenv("GOCD_URL"),
+			Username: os.Getenv("GOCD_USERNAME"),
+			Password: os.Getenv("GOCD_PASSWORD"),
+		}
+
+		c := cfg.Client()
+		p, _, _ := c.PipelineTemplates.Get(context.Background(), "test-pipeline-template")
```

This should test the error and return it, if we get an error rather than swallowing the error.